text/3463-crates-io-policy-update.md

```diff
@@ -0,0 +1,224 @@
+all the information needed for cargo to run the dependency resolution algorithm.
+
+In case the index dataset is insufficient for your purposes, we also publish a
+**database dump** every 24 hours. This includes the majority of data from our
+database except for sensitive private information. The latest database dump is
+available at <https://static.crates.io/db-dump.tar.gz> and information on using
+the content is contained in the tarball. Please note that while we aim to keep
+the data structure somewhat stable, we can not give any stability guarantees on
+the exact database table layouts.
+
+If the index repository and the database dump are insufficient you may also use
+the crates.io API directly, though it is at the discretion of the crates.io to
+block any excessive usage. We require users of the crates.io API to limit
+themselves to a maximum of 1 request per second.
+
+We also require all API users to provide a user-agent header that allows us to
+uniquely identify your application. This allows us to more accurately monitor
+any impact your application may have on our service. Providing a user agent that
+only identifies your HTTP client library (such as `request/0.9.1`) increases the
+likelihood that we will block your traffic.
+
+It is recommended, to include contact information in your user-agent header:
+
+- Bad: `User-Agent: reqwest/0.9.1`
+- Better: `User-Agent: my_bot`
+- Best: `User-Agent: my_bot (my_bot.com/info)` or `User-Agent: my_bot (help@my_bot.com)`
+
+This allows us to contact you if we would like a change in your application's
+behavior without having to block your traffic.
+
+We reserve the right to block traffic from any client that we determine to be in
+violation of this policy or causing an impact on the integrity of our service.
+
+
+## Security
+
+Safety is one of the core principles of Rust, and to that end, we would like to
+ensure that cargo and crates.io have secure implementations. To learn more about
+disclosing security vulnerabilities, please reference the
+[Rust Security policy](https://www.rust-lang.org/policies/security) for more
+details.
+
+Thank you for taking the time to responsibly disclose any issues you find.
+
+
+## Violations and Enforcement
+
+crates.io retains full discretion to take action in response to a violation of
+these policies, including account suspension, account termination, or removal of
+content.
+
+We will however not be proactively monitoring the site for these kinds of
+violations, but instead relying on the community to draw them to our attention.
+
+While the majority of interactions between individuals in the Rust community
+falls within our policies, violations of those policies do occur at times.
+When they do, the crates.io team may need to take enforcement action to address
+the violations. In all cases, these actions are permanent and there is no basis
+to reverse a moderation action taken by the crates.io team.
```

Comment:
I think that it's fair to say that content deletion is permanent, but I think that it's a bit harsh to assume that all actions are permanent. For example, if someone's account gets compromised, it makes sense to temporarily suspend their account until the original author regains control, but the implication here is that you will never get your account back if this happens.

Reply:
I agree. We usually lock the user accounts, but we don't have such a quarantine mechanism for crates yet. This part was copied over from the PyPI policies if I remember correctly. Do you have a suggestion on how to rephrase it?

Reply:
Sure, here's a quick draft.

> In all cases, content and account deletion is permanent and there is no basis to reverse these moderation actions taken by the crates.io team. Account suspension may be lifted at the team's discretion however, for example in the case of someone's account being compromised.

Reply:
Proper suggestion:

```suggestion
While the majority of interactions between individuals in the Rust community
falls within our policies, violations of those policies do occur at times.
When they do, the crates.io team may need to take enforcement action to address
the violations. In all cases, content and account deletion is permanent and there
is no basis to reverse these moderation actions taken by the crates.io team.
Account suspension may be lifted at the team's discretion however, for example in
the case of someone's account being compromised.
```

Reply:
added ✅